src/EPR.Calculator.Service.Function/Services/ErrorReportService.cs

```diff
@@ -82,11 +82,20 @@ public List<ErrorReport> HandleObligatedErrors(IEnumerable<CalculatorRunOrganisa
         {
             if (orgDetails == null) throw new ArgumentNullException(nameof(orgDetails));
 
-            var obligatedErrors = orgDetails
-                                    .Where(x => x.ObligationStatus == ObligationStates.Error)
-                                    .Select(x => CreateError(x.OrganisationId, x.SubsidiaryId, calculatorRunId, createdBy, x.ErrorCode ?? string.Empty));
+            return orgDetails
+                    .Where(x => (x.ObligationStatus == ObligationStates.Error))
+                    .Select(x => CreateError(x.OrganisationId, x.SubsidiaryId, calculatorRunId, createdBy, x.ErrorCode ?? string.Empty))
+                    .ToList();
+        }
+
+        public List<ErrorReport> HandleObligatedWarnings(IEnumerable<CalculatorRunOrganisationDataDetail> orgDetails, int calculatorRunId, string createdBy)
+        {
+            if (orgDetails == null) throw new ArgumentNullException(nameof(orgDetails));
 
-            return obligatedErrors.ToList();
+            return orgDetails
+                    .Where(x => x.ObligationStatus == ObligationStates.Obligated && !string.IsNullOrEmpty(x.ErrorCode))
+                    .Select(x => CreateError(x.OrganisationId, x.SubsidiaryId, calculatorRunId, createdBy, x.ErrorCode ?? string.Empty))
```

Comment:
we know it's not null
```suggestion
                    .Select(x => CreateError(x.OrganisationId, x.SubsidiaryId, calculatorRunId, createdBy, x.ErrorCode))
```